back/app/services/gotenberg_client.rb

```
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class GotenbergClient
+  FileTypeUnsupportedError = Class.new(StandardError)
+
+  def initialize
+    @api_url = ENV.fetch('GOTENBURG_PDF_URL', 'http://gotenberg:3000')
+  end
+
+  # Use Gotenberg web service (separate docker container) to render html to PDF
+  def render_html_to_pdf(html)
+    return false unless up?
+
+    payload = {
+      'preferCssPageSize' => true,
+      'generateDocumentOutline' => true,
+      'generateTaggedPdf' => true,
+      'index.html' => Faraday::Multipart::FilePart.new(
+        StringIO.new(html),
+        'text/html; charset=utf-8',
+        'index.html'
+      )
+    }
+    url = "#{@api_url}/forms/chromium/convert/html"
+
+    conn = Faraday.new(url) do |f|
+      f.request :multipart, flat_encode: true
+      f.adapter :net_http
+    end
+    response = conn.post(url, payload)
+
+    output_pdf = Tempfile.new(['gotenberg', '.pdf'])
+    output_pdf.binmode
+    output_pdf.write(response.body)
+    output_pdf.rewind
+
+    output_pdf
+  end
+
+  def render_libreoffice_to_pdf(file, content_type, file_name)
+    return false unless up?
```

(nitpicky) I see that you followed the existing pattern, but it doesn't feel quite right to check for health before sending the request and to silently ignore the error if it does not. I think if the service is configured (via the env variable), we should assume it's up and running, and let it fail if it doesn't.

---

I agree, I thought the same bit didn't take the effort to change it. Will do.